pkg/k8shandler/certificates.go

```diff
@@ -2,160 +2,98 @@ package k8shandler
 
 import (
 	"fmt"
-	"os/exec"
+	"io/ioutil"
+	"path"
+	"sync"
 
+	"github.com/openshift/cluster-logging-operator/pkg/certificates"
 	"github.com/openshift/cluster-logging-operator/pkg/constants"
 	"github.com/openshift/cluster-logging-operator/pkg/logger"
 	"github.com/openshift/cluster-logging-operator/pkg/utils"
 	"k8s.io/apimachinery/pkg/api/errors"
-
-	"k8s.io/apimachinery/pkg/util/sets"
 )
 
-var deprecatedKeys = sets.NewString("app-ca", "app-key", "app-cert", "infra-ca", "infra-key", "infra-cert")
+var mutex sync.Mutex
```

Comment:
Can you use RWMutex instead?

Reply:
@blockloop  Regarding the mutex: Isn't the Reconcile function executed in its own goroutine by the operator SDK?

Reply:
That I don't know. What would that change?